### PR TITLE
[WIP] Add actor checknis

### PR DESCRIPTION
**Pull request by mmuzila**
This actor checks configuration of ypbind and nsswitch. Address of NIS server cannot be specified by host name if NIS is used to resolve host names.

### CODE REVIEW

File: repos/system_upgrade/el7toel8/actors/checknis/actor.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    line = next(it)
+
+    if line.endswith("\\\n"):
+        line = line[:-2] + " " + get_line(it)
+
+    return line
+
+class CheckNis(Actor):
+    """
+    Check configuration of ypbind and nsswitch. Address of NIS server cannot be
```
Comment (vojtechsokol) [inline repos/system_upgrade/el7toel8/actors/checknis/actor.py]: Summary line and rest of the docstring should be separated by a blank line, see [Contribution guidelines](https://github.com/oamg/leapp-repository/blob/master/CONTRIBUTING.md#code-style) and [PEP 257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings).

File: repos/system_upgrade/el7toel8/actors/checknis/actor.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    tags = (IPUWorkflowTag, ChecksPhaseTag)
+
+
+    def nis_in_nsswitch_hosts(self):
+        """
+        Check if hosts are resolved by NIS (nsswitch.conf)
+        """
+
+        try:
+            fp = open("/etc/nsswitch.conf")
```
Comment (vojtechsokol) [inline repos/system_upgrade/el7toel8/actors/checknis/actor.py]:
```suggestion
            with open("/etc/nsswitch.conf") as f:
```

File: repos/system_upgrade/el7toel8/actors/checknis/actor.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            rawlines = fp.readlines()
+            fp.close()
+        except:
```
Comment (vojtechsokol) [inline repos/system_upgrade/el7toel8/actors/checknis/actor.py]: consider at least catching `OSError` only, or is there any reason to catch everything?
```suggestion
        except OSError:
```

**Comment by vinzenz** [comment]
Please apply the changes requested and rebase the PR to the current master.

File: repos/system_upgrade/el7toel8/actors/checknis/actor.py (hunk 3 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/actors/checknis/actor.py]: There's never a reason for such code to catch more than StandardError

**Comment by bocekm** [comment]
Hi @mmuzila, could you please update your actor based on the comments above? Thanks ;)

**Comment by pirat89** [comment]
@mmuzila Letting you know, we did several significant changes around reporting in the framework and provided API. Please, see other actors to see what changed or see thes [example](https://github.com/oamg/leapp/blob/master/docs/source/actor-rhel7-to-rhel8.md#producing-data-for-other-actors-and-reporting). Also please rebase against the current master.

**Comment by bocekm** [comment]
@mmuzila, how does it look like with this task?

**Comment by bocekm** [comment]
@mmuzila, ping :)

**Comment by bocekm** [comment]
Filip, I'm looking forward to the actor you're going to finish. We'll wait, I know you have another duties now - good luck with that!

**Pull request closed without merging**